test/linalg/dense.jl

```diff
@@ -582,6 +582,25 @@ end
     end
 end
 
+@testset "issue #23366 (Int Matrix to Int power)" begin
+    @testset "Tests for $elty" for elty in (Int128, Int16, Int32, Int64, Int8,
+                                            UInt128, UInt16, UInt32, UInt64, UInt8,
+                                            BigInt)
+        @test_throws DomainError elty[1 1;1 0]^-2
+        @test elty[1 1;1 0]^2 == elty[2 1;1 1]
```

Comment:
Perhaps use some `@inferred` here since thats what this PR is about? Something like
```julia
@test (@inferred elty[1 1; 1 0]^2) == elty[2 1; 1 1]
```
should do the trick :)